src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -788,6 +794,27 @@ static term gpiodriver_set_int(Context *ctx, int32_t target_pid, term cmd)
         return create_pair(ctx, ERROR_ATOM, globalcontext_make_atom(ctx->global, invalid_trigger_atom));
     }
 
+    if (term_get_tuple_arity(cmd) == 4) {
+        term pid = term_get_tuple_element(cmd, 3);
+        if (UNLIKELY(!term_is_pid(pid) && !term_is_atom(pid))) {
+            AVM_LOGE(TAG, "Invalid listener parameter, must be a pid() or registered process!");
+            return create_pair(ctx, ERROR_ATOM, invalid_listener_atom);
+        }
+        if (term_is_pid(pid)) {
+            target_local_pid = term_to_local_process_id(pid);
+        } else {
+            int pid_atom_index = term_to_atom_index(pid);
+            int32_t registered_process = (int32_t) globalcontext_get_registered_process(ctx->global, pid_atom_index);
+            if (UNLIKELY(registered_process == 0)) {
+                AVM_LOGE(TAG, "Invalid listener parameter, atom() is not a registered process name!");
+                return create_pair(ctx, ERROR_ATOM, NOPROC_ATOM);
```

Comment:
I think this is not coherent with the ESP32 version return value. Can you check it?

Reply:
The error return values for stm32 have already been updated to return `{error, Reason}`, the esp32 gpio diver has not been updated yet, but a TODO comment had been left that the driver needs to be updated, I just felt that was outside the scope of this PR to make those changes now.